server/sonar-server/src/main/java/org/sonar/server/rule/RegisterRules.java

```diff
@@ -349,7 +344,8 @@ private static boolean mergeTags(RulesDefinition.Rule ruleDef, RuleDto dto) {
     if (RuleStatus.REMOVED == ruleDef.status()) {
       dto.setSystemTags(Collections.<String>emptySet());
       changed = true;
-    } else if (!dto.getSystemTags().containsAll(ruleDef.tags())
+    } else if (dto.getSystemTags().size() != ruleDef.tags().size() ||
```

Comment:
do we know that `dto.getSystemTags()` and `ruleDef.tags()` have the same equivalence order? Both are HashSet?